exporter/signalfxexporter/translation/translator_test.go

```diff
@@ -2073,3 +2085,214 @@ func TestDimensionsEqual(t *testing.T) {
+	}
+}
+
+func TestDeltaMetricInt(t *testing.T) {
+	const delta1 = 7
+	const delta2 = 11
+	md1 := intMD(10, 0)
+	md2 := intMD(20, delta1)
+	md3 := intMD(30, delta1+delta2)
+	pts1, pts2 := requireDeltaMetricOk(t, md1, md2, md3)
+	for _, pt := range pts1 {
+		require.EqualValues(t, delta1, *pt.Value.IntValue)
+	}
+	for _, pt := range pts2 {
+		require.EqualValues(t, delta2, *pt.Value.IntValue)
+	}
+}
+
+func TestNegativeDeltas(t *testing.T) {
+	md1 := intMD(10, 0)
+	md2 := intMD(20, 13)
+	md3 := intMDAfterReset(30, 5)
+	pts1, pts2 := requireDeltaMetricOk(t, md1, md2, md3)
+	for _, pt := range pts1 {
+		require.EqualValues(t, 13, *pt.Value.IntValue)
+	}
+	for _, pt := range pts2 {
+		// since the counter went down (assuming a reset), we expect a delta of the most recent value
+		require.EqualValues(t, 5, *pt.Value.IntValue)
+	}
+}
+
+func TestDeltaTranslatorNoMatchingMapping(t *testing.T) {
+	c := converter(t, map[string]string{"foo": "bar"})
+	md := intMD(1, 1)
+	pts, _ := c.MetricDataToSignalFxV2([]consumerdata.MetricsData{md})
+	idx := indexPts(pts)
+	require.Equal(t, 1, len(idx))
+}
+
+func TestDeltaTranslatorMismatchedValueTypes(t *testing.T) {
+	c := converter(t, map[string]string{"system.cpu.time": "system.cpu.delta"})
+	md1 := baseMD()
+	md1.Metrics[0].Timeseries = []*metricspb.TimeSeries{
+		intTS("cpu0", "user", 1, 1, 1),
+	}
+	_, _ = c.MetricDataToSignalFxV2([]consumerdata.MetricsData{md1})
+	md2 := baseMD()
+	md2.Metrics[0].Timeseries = []*metricspb.TimeSeries{
+		dblTS("cpu0", "user", 1, 1, 1),
+	}
+	pts, _ := c.MetricDataToSignalFxV2([]consumerdata.MetricsData{md2})
+	idx := indexPts(pts)
+	require.Equal(t, 1, len(idx))
+}
+
+func requireDeltaMetricOk(t *testing.T, md1, md2, md3 consumerdata.MetricsData) (
+	[]*sfxpb.DataPoint, []*sfxpb.DataPoint,
+) {
+	c := converter(t, map[string]string{"system.cpu.time": "system.cpu.delta"})
+
+	dp1, dropped1 := c.MetricDataToSignalFxV2([]consumerdata.MetricsData{md1})
+	require.Equal(t, 0, dropped1)
+	m1 := indexPts(dp1)
+	require.Equal(t, 1, len(m1))
+
+	dp2, dropped2 := c.MetricDataToSignalFxV2([]consumerdata.MetricsData{md2})
+	require.Equal(t, 0, dropped2)
+	m2 := indexPts(dp2)
+	require.Equal(t, 2, len(m2))
+
+	origPts, ok := m2["system.cpu.time"]
+	require.True(t, ok)
+
+	deltaPts1, ok := m2["system.cpu.delta"]
+	require.True(t, ok)
+	require.Equal(t, len(origPts), len(deltaPts1))
+	counterType := sfxpb.MetricType_GAUGE
+	for _, pt := range deltaPts1 {
+		require.Equal(t, &counterType, pt.MetricType)
+	}
+
+	dp3, dropped3 := c.MetricDataToSignalFxV2([]consumerdata.MetricsData{md3})
+	require.Equal(t, 0, dropped3)
+	m3 := indexPts(dp3)
+	require.Equal(t, 2, len(m3))
+
+	deltaPts2, ok := m3["system.cpu.delta"]
+	require.True(t, ok)
+	require.Equal(t, len(origPts), len(deltaPts2))
+	for _, pt := range deltaPts2 {
+		require.Equal(t, &counterType, pt.MetricType)
+	}
+	return deltaPts1, deltaPts2
+}
+
+func converter(t *testing.T, mapping map[string]string) *MetricsConverter {
```

Comment:
nit: same comment as for `metricName` . can we rename it to `testConverter`? I'd do similar for other func/var names in this file

Reply:
Fixed. I wasn't able to come up with names for the other test helper functions that were clear, maybe because the `test` function name prefix sound like a verb.